kyo-data/shared/src/main/scala/kyo/Schedule.scala

```diff
@@ -0,0 +1,348 @@
+    def repeat(n: Int): Schedule =
+        immediate.repeat(n)
+
+    /** Creates a schedule with a fixed interval between iterations.
+      *
+      * @param interval
+      *   the fixed interval
+      * @return
+      *   a new schedule with the specified fixed interval
+      */
+    def fixed(interval: Duration): Schedule = Fixed(interval)
+
+    /** Creates a schedule with linearly increasing intervals.
+      *
+      * @param base
+      *   the initial interval
+      * @return
+      *   a new schedule with linearly increasing intervals
+      */
+    def linear(base: Duration): Schedule =
+        if base == Duration.Zero then immediate.forever
+        else Linear(base)
+
+    /** Creates a schedule with intervals following the Fibonacci sequence.
+      *
+      * @param a
+      *   the first interval
+      * @param b
+      *   the second interval
+      * @return
+      *   a new schedule with Fibonacci sequence intervals
+      */
+    def fibonacci(a: Duration, b: Duration): Schedule =
+        if a == Duration.Zero && b == Duration.Zero then immediate.forever
+        else Fibonacci(a, b)
+
+    /** Creates a schedule with exponentially increasing intervals.
+      *
+      * @param initial
+      *   the initial interval
+      * @param factor
+      *   the factor by which to increase the interval
+      * @return
+      *   a new schedule with exponentially increasing intervals
+      */
+    def exponential(initial: Duration, factor: Double): Schedule =
+        if initial == Duration.Zero then immediate
+        else if factor == 1.0 then fixed(initial)
+        else Exponential(initial, factor)
+
+    /** Creates a schedule with exponential backoff and a maximum delay.
+      *
+      * @param initial
+      *   the initial interval
+      * @param factor
+      *   the factor by which to increase the interval
+      * @param maxBackoff
+      *   the maximum delay allowed
+      * @return
+      *   a new schedule with exponential backoff and a maximum delay
+      */
+    def exponentialBackoff(initial: Duration, factor: Double, maxBackoff: Duration): Schedule =
+        if initial == Duration.Zero then immediate
+        else if factor == 1.0 then fixed(initial)
+        else ExponentialBackoff(initial, factor, maxBackoff)
+
+    private[kyo] object internal:
+
+        case object Immediate extends Schedule:
+            val next = Maybe((Duration.Zero, Done))
+            def show = "Schedule.immediate"
+
+        case object Never extends Schedule:
+            def next = Maybe.empty
+            def show = "Schedule.never"
+
+        case object Done extends Schedule:
+            def next = Maybe.empty
+            def show = "Schedule.done"
+
+        final case class Fixed(interval: Duration) extends Schedule:
+            val next = Maybe((interval, this))
+            def show = s"Schedule.fixed(${interval.show})"
+
+        final case class Exponential(initial: Duration, factor: Double) extends Schedule:
```

Comment:
I think it's ok in terms of binary compatibility since these are internal APIs. The goal of the different classes is for `Schedule` to automatically reduce to one of these canonical representations that can be compared for equality given that they're case classes. These tests show the functionality: https://github.com/getkyo/kyo/pull/733/files#diff-66ce2c17e23c73ff33f5e658a7ba3ef1972eb58d007873d50d795dae21a8e52dR578. I'm planning to use this behavior to optimize the execution of a new `Timer` implementation based on `Schedule`.